debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -161,10 +162,18 @@ public Object getValue(String columnName, TypeMetadataImpl typeMetadata, Value r
         if (typeMetadata.isArray()) {
             try {
                 final String dataString = rawValue.asString();
-                PgArray arrayData = new PgArray(connection.get(), connection.get().getTypeInfo().getPGArrayType(typeMetadata.getFullType()), dataString);
-                Object deserializedArray = arrayData.getArray();
+                int arrayTypeOid = connection.get().getTypeInfo().getPGArrayType(typeMetadata.getFullTypeWithSchema());
+                PgArray arrayData = new PgArray(connection.get(), arrayTypeOid, dataString);
```

Comment:
@jpechane I get a "connection is closed" error here — is there a better way to get an active connection?

Reply:
This should happen only when the `RecordsStreamProducer` is closed.

Reply:
Yeah, I'm consistently getting it in the `shouldReceiveChangesForInsertsWithPostgisArrayTypes` test under both decoderbufs+wal2json. That's the only remaining issue AFAICT.